### PR TITLE
feat: added modal component

### PR DESCRIPTION
**Pull request by Pratik1005**

### CODE REVIEW

**Review by JowelTisso — APPROVED** [review]
Looks Good.

File: components/modal/modal.css [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    background-color: rgba(0,0,0,0.5);
+    display: grid;
+    place-items: center;
+    visibility: hidden;
+    z-index: -10; 
+}
+
+.open-modal {
+    visibility: visible;
+    z-index: 10;
```
Comment (AnjaliDeshwani) [inline components/modal/modal.css:16]: Just curious to know why you have used large z-index:10? It could have been achieved by z-index:1 or 2.

**Review by AnjaliDeshwani — APPROVED** [review]
Looks good to me.

File: components/modal/modal.css (hunk 1 above)
Comment (Pratik1005) [reply components/modal/modal.css:16]: Yes I can use small z-index. It has become a habit to use large z-index for overlay. I will improve on this for next projects.

**Pull request merged**